lib/widgets/matrix.dart

```diff
@@ -107,14 +120,16 @@ class MatrixState extends State<Matrix>
   late String currentClientSecret;
   RequestTokenResponse? currentThreepidCreds;
 
-  void setActiveClient(Client? cl) {
-    final i = widget.clients.indexWhere((c) => c == cl);
-    if (i != -1) {
-      _activeClient = i;
+  void setActiveClient(Client? newClient) {
+    _checkHomeserverExists(newClient);
+    final index = widget.clients.indexWhere((client) => client == newClient);
+    if (index != -1) {
+      _activeClient = index;
       // TODO: Multi-client VoiP support
       createVoipPlugin();
+      _storePersistActiveAccount(newClient!);
     } else {
-      Logs().w('Tried to set an unknown client ${cl!.userID} as active');
+      Logs().w('Tried to set an unknown client ${newClient!.userID} as active');
```

Comment:
What happens in this case? Chat list will change the active client, but this not.